cordapp-example/contracts-kotlin/src/main/kotlin/com/example/state/CashState.kt

```diff
@@ -11,12 +11,13 @@ import net.corda.core.schemas.PersistentState
 import net.corda.core.schemas.QueryableState
 
 @BelongsToContract(CashContract::class)
-data class CashState(val value: Long, override val owner: AbstractParty): OwnableState, QueryableState {
+data class CashState(val value: Long, val creator: AbstractParty, override val owner: AbstractParty): OwnableState, QueryableState {
     override val participants: List<AbstractParty> get() = listOf(owner)
 
     override fun generateMappedObject(schema: MappedSchema): PersistentState {
         return when (schema) {
             is CashSchemaV1 -> CashSchemaV1.PersistentCash(
+                    this.creator.nameOrNull().toString(),
```

Comment:
I think in states (same as in contracts) we don't have access to serviceHub. I couldn't find a way to check party.
I tried to create identity with empty name, country, location and I couldn't. Even if I could, this state will not be included in valid transaction, because contract will fail, so this `nameOrNull` should not be a problem in our case.

Reply:
`AnonymousParty` extends `AbstractParty`, that would be valid, and have no name.
Is it possible to save the public key instead?